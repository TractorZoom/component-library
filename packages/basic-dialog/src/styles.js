import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    contentContainer: {
        padding: `0 ${theme.spacing(4)}px`,
    },
    largeContainer: {
        minHeight: '80%',
        minWidth: '80%',
    },
    smallContainer: {
        minWidth: '50%',
    },
    titleContainer: {
        alignItems: 'center',
        display: 'flex',
        justifyContent: 'space-between',
        padding: theme.spacing(4),
    },
    footerContainer: {
        padding: theme.spacing(4),
    },
}));

export default useStyles;
