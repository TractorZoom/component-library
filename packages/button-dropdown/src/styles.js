import { makeStyles } from '@material-ui/core/styles';

const loadingSpinner = {
    height: '14px !important',
    width: '14px !important',
};

const useStyles = makeStyles((theme) => ({
    loadingSpinner: {
        ...loadingSpinner,
        marginRight: theme.spacing(1),
    },
}));

export default useStyles;
