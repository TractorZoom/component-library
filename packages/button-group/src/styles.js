import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    button: {
        '&:not(:last-child)': {
            marginRight: theme.spacing(2),
        },
    },
    buttonGroup: {
        display: 'inline-block',
        overflowX: 'scroll',
        width: '100%',
        whiteSpace: 'nowrap',
    },
}));

export default useStyles;
