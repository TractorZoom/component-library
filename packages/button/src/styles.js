import { makeStyles } from '@material-ui/core/styles';

const loadingSpinner = {
    height: '24px !important',
    width: '24px !important',
};

const useStyles = makeStyles((theme) => ({
    loadingSpinnerInIcon: {
        ...loadingSpinner,
    },
    loadingSpinnerInLabel: {
        ...loadingSpinner,
        marginRight: theme.spacing(1),
    },
}));

export default useStyles;
