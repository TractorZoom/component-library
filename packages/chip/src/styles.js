import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    active: {
        backgroundColor: `${theme.palette.primary.main}1F !important`,
        color: `${theme.palette.primary.main} !important`,
    },
    chip: {
        backgroundColor: 'rgba(0, 0, 0, 0.05)',
        '&:hover': {
            backgroundColor: 'rgba(0, 0, 0, 0.08)',
        },
    },
}));

export default useStyles;
