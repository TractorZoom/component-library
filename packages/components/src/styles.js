import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    lozenge: {
        paddingBottom: 2,
        paddingTop: 2,
        fontWeight: 600,
        fontSize: 10,
        borderRadius: 2,
        padding: '1px 2px 1px 2px',
    },
}));

export default useStyles;
