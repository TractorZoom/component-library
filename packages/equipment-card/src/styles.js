import { makeStyles } from '@material-ui/core/styles';

const iconButton = (theme) => ({
    padding: theme.spacing(1),
    position: 'absolute',
    right: 18,
    top: 176,
});

export default makeStyles((theme) => ({
    auctionDetailsText: {
        fontSize: 10,
        fontWeight: 600,
    },
    canvas: {
        display: 'none',
    },
    cardContent: {
        padding: theme.spacing(5),
        paddingBottom: theme.spacing(2),
    },
    cardHolder: {
        position: 'relative',
    },
    checkedButton: {
        ...iconButton(theme),
        '&:hover': {
            backgroundColor: theme.palette.success.main,
        },
        backgroundColor: theme.palette.success.main,
        color: theme.palette.common.white,
        'pointer-events': 'all',
    },
    details: {
        alignItems: 'center',
        display: 'flex',
        fontSize: theme.spacing(4),
        justifyContent: 'space-between',
        marginBottom: theme.spacing(-1),
    },
    detailsText: {
        color: '#4C4C4C',
        fontSize: 12,
        marginTop: theme.spacing(2),
    },
    divider: {
        background: '#CFD2D7',
    },
    locationAndSaleDate: {
        display: 'flex',
        justifyContent: 'space-between',
        marginTop: 8,
    },
    lozenge: {
        paddingBottom: 2,
        paddingTop: 2,
        color: '#045C3E',
        backgroundColor: '#DFFCED',
        fontWeight: 600,
        fontSize: 10,
        borderRadius: 2,
        padding: '1px 2px 1px 2px',
    },
    makeModelTitle: {
        fontSize: 14,
    },
    media: {
        height: 200,
    },
    price: {
        color: theme.palette.success.main,
        fontSize: theme.spacing(4),
        fontSize: 22,
        fontWeight: 500,
    },
    primaryDetail: {
        fontSize: 22,
        fontWeight: 500,
    },
    root: {
        border: '1px solid #CFD2D7',
        borderRadius: 5,
        marginBottom: theme.spacing(2),
        minWidth: 275,
        '&:hover': {
            borderColor: '#BEBEBE',
        },
    },
    secondaryDetail: {
        color: theme.palette.text.secondary,
        fontSize: 14,
        fontWeight: 500,
        paddingBottom: 10,
    },
    selectButton: {
        ...iconButton(theme),
        '&:hover': {
            backgroundColor: theme.palette.common.white,
            color: '#184EAC',
        },
        backgroundColor: theme.palette.common.white,
        border: '1px solid #CFD2D7',
        'pointer-events': 'all',
    },
    selectedCard: {
        border: `3px solid ${theme.palette.success.main}`,
        '&:hover': {
            borderColor: theme.palette.success.main,
        },
    },
    variableDetail: {
        fontSize: theme.spacing(4),
        fontWeight: 'bold',
    },
}));
