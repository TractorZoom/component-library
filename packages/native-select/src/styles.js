import { makeStyles } from '@material-ui/core/styles';

export default (darkMode) =>
    makeStyles((theme) => ({
        inputBorder: darkMode ? { borderColor: theme.palette.common.white } : {},
        inputLabel: darkMode
            ? {
                  '& label': {
                      color: theme.palette.common.white,
                  },
                  '& svg': {
                      color: theme.palette.common.white,
                  },
              }
            : {},
        inputValue: darkMode ? { color: theme.palette.common.white } : {},
    }));
