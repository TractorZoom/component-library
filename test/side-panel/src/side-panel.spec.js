import { render, screen } from '@testing-library/react';
import SidePanel from '../../../packages/side-panel/src';
import React from 'react';
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles';

describe('side panel', () => {
    it('should render the side panel empty', () => {
        // given
        const theme = createMuiTheme();
        const props = {
            children: [],
            open: true,
            width: 'md',
            setOpen: () => {},
        };

        // when
        render(
            <ThemeProvider theme={theme}>
                <SidePanel {...props} />
            </ThemeProvider>
        );

        // then
        expect(screen).toMatchSnapshot();
    });

    it('should render the side panel with a child', () => {
        //given
        const theme = createMuiTheme();
        const props = {
            children: [<div>Child</div>],
            open: true,
            width: 'md',
            setOpen: () => {},
        };

        //when
        render(
            <ThemeProvider theme={theme}>
                <SidePanel {...props} />
            </ThemeProvider>
        );

        //then
        expect(screen).toMatchSnapshot();
    });

    it('should render the side panel with a header component', () => {
        //given
        const theme = createMuiTheme();
        const props = {
            children: [<div>Child</div>],
            headerComponent: <div>Header Component</div>,
            open: true,
            width: 'md',
            setOpen: () => {},
        };

        //when
        render(
            <ThemeProvider theme={theme}>
                <SidePanel {...props} />
            </ThemeProvider>
        );

        //then
        expect(screen).toMatchSnapshot();
    });
});
