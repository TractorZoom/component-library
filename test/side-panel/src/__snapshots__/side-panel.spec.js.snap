// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Panel should render the side panel empty 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
>
  <div
    className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-4 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="makeStyles-drawerHeader-2"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      className="MuiDivider-root makeStyles-headerDivider-6"
    />
  </div>
</div>
`;

exports[`Side Panel should render the side panel with a child 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
>
  <div
    className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-4 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="makeStyles-drawerHeader-2"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      className="MuiDivider-root makeStyles-headerDivider-6"
    />
    <div
      isOpen={true}
      setOpen={[Function]}
    >
      Child
    </div>
  </div>
</div>
`;
